pkg/worker/worker.go

```diff
@@ -61,8 +62,14 @@ func GatherResults(waitfile string, url string) error {
 	}
 
 	// transmit back the results file.
-	return DoRequest(url, func() (io.Reader, error) {
-		outfile, err := os.Open(s)
-		return outfile, errors.WithStack(err)
+	err = DoRequest(url, func() (io.Reader, error) {
+		outfile, e2 := os.Open(s)
+		return outfile, errors.WithStack(e2)
```

Comment:
I'm not sure about this; outfile on line 66 is in a different scope to line 39, so on line 70, outfile will always be nil.

I suggest

s/outfile/f here to make it clear that you're talking about a different variable.